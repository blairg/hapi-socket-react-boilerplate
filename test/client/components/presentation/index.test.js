/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/extensions */

import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Todos from './../../../../src/client/components/containers/todos.jsx';
import Index from './../../../../src/client/components/presentation/index.jsx';

const middlewares = [];
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

describe('client/components/presentation/index -> <Index todos={todos} />', () => {
  test('should contain body and title in the Index component', () => {
    const todo = { title: 'my title', body: 'my body' };

    const initialState = { setTodos: { todos: [todo] } };
    const store = mockStore(initialState);

    const todosComponent = (
      <Provider store={store}>
        <Todos entries={[todo]} />
      </Provider>
    );
    const wrapper = Enzyme.shallow(<Index todos={todosComponent} />);

    expect(wrapper.html().includes(todo.title)).toBeTruthy();
    expect(wrapper.html().includes(todo.body)).toBeTruthy();
  });
});
