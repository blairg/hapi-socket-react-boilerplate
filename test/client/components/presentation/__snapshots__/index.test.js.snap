// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `client/components/presentation/index -> <Index todos={todos} /> should render correctly 1`
] = `
<html
  lang="en"
>
  <head>
    <meta
      charSet="UTF-8"
    />
    <title>
      Node.js + Hapi.js + Socket.io + HTTPS + HTTP/2 + React.js + Redux.js | Todo Example
    </title>
    <link
      as="style"
      href="/css/vendors.css"
      rel="preload"
    />
    <link
      as="style"
      href="/css/bundle.css"
      rel="preload"
    />
    <link
      as="script"
      href="/js/bundle.js"
      rel="preload"
    />
    <link
      href="/css/vendors.css"
      rel="stylesheet"
      type="text/css"
    />
    <link
      href="/css/bundle.css"
      rel="stylesheet"
      type="text/css"
    />
  </head>
  <body>
    <div
      id="app"
    >
      [{"title":"my title","body":"my body"}]
    </div>
    <script
      async={true}
      src="/js/bundle.js"
    />
  </body>
</html>
`;
